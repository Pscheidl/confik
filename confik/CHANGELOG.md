# Changelog

## Unreleased

- Attributes that receive expressions (`default` and `from`) now need to be unquoted. E.g.:

  ```diff
  - struct MyStruct(#[config(default = "\"Hello World\"") String)
  + struct MyStruct(#[config(default = "Hello World") String)

  - struct MyStruct(#[config(default = "5usize") usize)
  + struct MyStruct(#[config(default = 5usize) usize)
  ```

## 0.7.0

- Initial release.
