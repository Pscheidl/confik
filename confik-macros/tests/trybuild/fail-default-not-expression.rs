use confik::Configuration;

#[derive(Debug, Configuration, PartialEq, Eq)]
struct Config {
    #[confik(default = +++4_u32)]
    param: u32,
}

fn main() {}
