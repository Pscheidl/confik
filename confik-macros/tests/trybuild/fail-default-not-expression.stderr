error: expected expression
 --> tests/trybuild/fail-default-not-expression.rs:5:24
  |
5 |     #[confik(default = +++4_u32)]
  |                        ^
